tree/ntuple/v7/src/RNTupleJoinTable.cxx

```diff
@@ -104,17 +104,8 @@ void ROOT::Experimental::Internal::RNTupleJoinTable::Build(RPageSource &pageSour
    fIsBuilt = true;
 }
 
-ROOT::NTupleSize_t
-ROOT::Experimental::Internal::RNTupleJoinTable::GetFirstEntryNumber(const std::vector<void *> &valuePtrs) const
-{
-   const auto entryIndices = GetAllEntryNumbers(valuePtrs);
-   if (!entryIndices)
-      return ROOT::kInvalidNTupleIndex;
-   return entryIndices->front();
-}
-
-const std::vector<ROOT::NTupleSize_t> *
-ROOT::Experimental::Internal::RNTupleJoinTable::GetAllEntryNumbers(const std::vector<void *> &valuePtrs) const
+std::vector<ROOT::NTupleSize_t>
+ROOT::Experimental::Internal::RNTupleJoinTable::GetEntryIndexes(const std::vector<void *> &valuePtrs) const
```

Comment:
I'm not getting it from the commit message: why are we now returning a copy of the array rather than a pointer?

Reply:
Ah right, I will add it to the commit message. It's anticipatory right now for some larger changes to the join table I am working on. See [this comment](https://github.com/root-project/root/pull/17782#issuecomment-2673956270) for the gist of it. With these partitions, it may be the case that join values are present in more than one partition, and in this case their corresponding entry indexes will need to be accumulated somewhere, so it won't be possible to directly return a pointer anymore.

I can postpone changing the signature to the point where this is actually needed if you prefer, but I figured that since we're refactoring this method anyways, might as well do it here anyways.